modules/library/http/src/main/java/org/geotools/http/AbstractHttpClient.java

```diff
@@ -90,6 +103,22 @@ public boolean isTryGzip() {
         return tryGzip;
     }
 
+    protected static URL appendURL(URL oldUrl, String appendQuery) throws MalformedURLException {
+        String oldQuery = oldUrl.getQuery();
+        String newQuery = oldQuery != null ? oldQuery + "&" + appendQuery : appendQuery;
+
+        String newUrlString =
+                oldUrl.getProtocol() + "://" + oldUrl.getAuthority() + oldUrl.getPath();
+        if (newQuery != null && !newQuery.isEmpty()) {
+            newUrlString += "?" + newQuery;
+        }
+        if (oldUrl.getRef() != null) {
+            newUrlString += "#" + oldUrl.getRef();
+        }
+
+        return new URL(newUrlString);
```

Comment:
This seems to be missing the TCP port.
Also please use one of the dedicated URL constructors such as https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/URL.html#%3Cinit%3E(java.lang.String,java.lang.String,int,java.lang.String)
It prevents a lot of - possibly error prone - String juggling

Reply:
Done

Reply:
getAuthority returns both the hostname and port, but I have separated them to be more clear.